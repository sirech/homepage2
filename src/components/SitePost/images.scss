@import '../../layouts/colors';
@import '../../layouts/fonts';

.figure {
  width: 100%;

  @media only screen and (min-width: 500px) {
    &--left,
    &--right {
      margin-left: auto;
      margin-right: auto;
      max-width: 315px;
    }

    &--left {
      float: left;
      margin-right: 1rem;
    }

    &--right {
      float: right;
      margin-left: 1rem;
    }
  }

  img {
    border: 1px solid $gray-700;
  }

  &--no-border img {
    border: 0;
  }

  &__image-wrap {
    border: 1px solid $gray-700;
    margin-bottom: 0;
  }

  &__image-link {
    border: 0;
  }

  &__caption {
    border-bottom: 1px solid $gray-700;
    font-size: $font-size-xs;
    margin-top: 0.5rem;
    padding: 0 0 0.5rem;

    blockquote & {
      border-bottom: 0;
    }

    &:empty {
      display: none;
    }
  }

  &__attribution {
    display: block;
    letter-spacing: 0.05rem;
    text-transform: uppercase;
  }
}
