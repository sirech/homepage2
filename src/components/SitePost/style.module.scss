@import '../../layouts/colors';
@import '../../layouts/fonts';

.articles {
  padding-top: 2rem;
}

.article {
  padding-bottom: 0.5rem;

  header {
    border-bottom: 1px solid $gray-200;
    margin-bottom: 1rem;
    padding-bottom: 0.5rem;
    word-break: break-word;

    h1 {
      color: $gray-600;
    }

    time {
      color: $gray-500;
      display: inline;
      font-size: $font-size-xs;
      margin-right: 0.5rem;
    }
  }
}

.pageContent {
  padding-bottom: 0.5rem;

  h2 {
    border-bottom: 3px solid $gray-400;
    border-top: 3px solid $gray-400;

    margin-top: 2rem;
    padding: 0.5rem 0;
  }

  h3 {
    border-bottom: 2px solid $gray-400;
    margin-top: 1.5rem;
    padding-bottom: 0.5rem;
  }

  h4 {
    border-bottom: 1px solid $gray-400;
    padding-bottom: 0.5rem;
  }

  p {
    margin: 1rem 0;
  }

  blockquote {
    border-left: 10px solid $gray-800;
    box-shadow: rgba(0, 0, 0, 0.2) 0 0 5px 0;
    font-style: italic;
    padding: 0.25rem 1.5rem;
  }
}

:global(.gatsby-highlight) {
  code {
    font-size: 0.8rem;
    white-space: pre-wrap;
  }
}

:global(.guide) {
  background-color: $white;
  border: 1px solid map-get($theme-colors, 'light');
  border-radius: 4px;
  box-shadow: 0 2px 1px -1px rgba(0, 0, 0, 0.2), 0 1px 1px 0 rgba(0, 0, 0, 0.14),
    0 1px 3px 0 rgba(0, 0, 0, 0.12);

  margin-bottom: 1rem;
  padding: 1rem;

  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;

  ul {
    list-style: none;
    margin-bottom: 0;
  }

  h3 {
    margin-top: 0;
  }
}
