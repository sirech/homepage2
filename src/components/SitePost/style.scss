@import '../../layouts/colors.scss';
@import '../../layouts/fonts.scss';

.articles {
  padding-top: 2rem;

  article {
    padding-bottom: 0.5rem;

    header {
      padding-bottom: 0.5rem;
      margin-bottom: 1rem;
      border-bottom: 1px solid $gray-200;
      word-break: break-word;

      h1 {
        color: $gray-600;
      }

      .badge {
        margin-right: 0.2rem;
      }

      time {
        font-size: $font-size-xs;
        color: $gray-500;
        display: inline;
        margin-right: 0.5rem;
      }
    }

    .page-content {
      padding-bottom: 0.5rem;

      h2 {
        padding-top: 0.5rem;
        margin-bottom: 1rem;
      }

      p {
        margin: 1rem 0;
      }

      blockquote {
        background-color: $gray-100;
        border-left: 5px solid $gray-200;
        padding: 0.25rem 1.5rem;

        a {
          display: block;
          word-break: break-word;
        }
      }
    }
  }

  .gatsby-highlight {
    code {
      font-size: 0.8rem;
      white-space: pre-wrap;
    }
  }

  .guide {
    background-color: $gray-100;
    border: 1px solid $gray-300;
    border-radius: 15px;

    padding: 1rem;

    ul {
      margin-bottom: 0;
    }
  }
}
