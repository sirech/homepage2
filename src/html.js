import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'

let stylesStr
if (process.env.NODE_ENV === 'production') {
  try {
    // eslint-disable-next-line
    stylesStr = require('!raw-loader!../public/styles.css')
  } catch (e) {
    console.log(e)
  }
}

class HTML extends React.Component {
  render() {
    const head = Helmet.rewind()
    let css
    if (process.env.NODE_ENV === 'production') {
      css = (
        <style
          id="gatsby-inlined-css"
          dangerouslySetInnerHTML={{ __html: stylesStr }}
        />
      )
    }

    return (
      <html lang="en">
        <head>
          <meta charSet="utf-8" />
          <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
          <meta
            name="viewport"
            content="width=device-width, initial-scale=1.0"
          />
          {this.props.headComponents}
          {css}
          <link
            href="/img/apple-touch-icon.png"
            rel="apple-touch-icon"
            sizes="180x180"
          />
          <link href="/img/favicon.ico" rel="icon" type="image/x-icon" />
        </head>
        <body itemScope itemType="http://schema.org/WebPage">
          <div
            id="___gatsby"
            dangerouslySetInnerHTML={{ __html: this.props.body }}
          />
          {this.props.postBodyComponents}
        </body>
      </html>
    )
  }
}

HTML.propTypes = {
  body: PropTypes.string.isRequired,
  headComponents: PropTypes.node,
  postBodyComponents: PropTypes.node,
}

export default HTML
