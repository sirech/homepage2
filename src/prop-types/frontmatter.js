import PropTypes from 'prop-types'

export default PropTypes.shape({
  categories: PropTypes.arrayOf(PropTypes.string).isRequired,
  date: PropTypes.string.isRequired,
  layout: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  draft: PropTypes.bool.isRequired,
  description: PropTypes.string,
}).isRequired
