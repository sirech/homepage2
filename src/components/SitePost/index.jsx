import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'

import Link from 'gatsby-link'
import Container from '../Container'
import Categories from '../Categories'
import Share from '../Share'

import styles from './style.module.scss'
import './images.scss'

import frontmatterType from '../../prop-types/frontmatter'
import siteType from '../../prop-types/site'

class SitePost extends React.Component {
  render() {
    const { data, site } = this.props
    const { frontmatter, html } = data
    const { title, path, date, categories } = frontmatter

    return (
      <Container>
        <div className={styles.articles}>
          <article key={path} className={cx(`${styles.article}`)}>
            <header>
              <Link style={{ boxShadow: 'none' }} to={path}>
                <h1>{title}</h1>
                <time dateTime={date}>{date}</time>
              </Link>
              <Categories categories={categories} />
            </header>
            <section
              className={cx(`${styles.pageContent}`, 'clearfix')}
              dangerouslySetInnerHTML={{
                __html: html,
              }}
            />

            <section>
              <Share site={site} path={path} />
            </section>
          </article>
        </div>
      </Container>
    )
  }
}

SitePost.propTypes = {
  data: PropTypes.shape({
    frontmatter: frontmatterType,
    html: PropTypes.string.isRequired,
  }).isRequired,
  site: siteType,
}

export default SitePost
