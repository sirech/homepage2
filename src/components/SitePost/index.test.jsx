import React from 'react'
import renderer from 'react-test-renderer'
import SitePost from './index'

import frontmatter from '../../fixtures/frontmatter'
import html from '../../fixtures/html'

describe('components', () => {
  describe('SitePost', () => {
    it('renders correctly', () => {
      const data = { html: html(), frontmatter: frontmatter() }
      const component = renderer.create(<SitePost data={data} />)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
