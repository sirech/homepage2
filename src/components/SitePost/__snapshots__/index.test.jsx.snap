// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components SitePost renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="articles"
  >
    <article
      className="article"
    >
      <header>
        <a
          href="/the-post"
          style={
            Object {
              "boxShadow": "none",
            }
          }
        >
          <h1>
            My Post
          </h1>
          <time
            dateTime="2018/03/01"
          >
            2018/03/01
          </time>
        </a>
        <span
          className="badge badge-primary text-white mr-1"
        >
          JavaScript
        </span>
        <span
          className="badge badge-primary text-white mr-1"
        >
          React
        </span>
      </header>
      <section
        className="pageContent clearfix"
        dangerouslySetInnerHTML={
          Object {
            "__html": "
<div>
  <span>Header</span>
</div>
",
          }
        }
      />
      
    </article>
  </div>
</div>
`;
